src/asm/fstack.c

```diff
@@ -34,6 +34,9 @@ struct Context {
 	uint32_t uniqueID;
 	struct MacroArgs *macroArgs; /* Macro args are *saved* here */
 	uint32_t nbReptIters;
+	int32_t foreachValue;
+	int32_t foreachStep;
```

Comment:
Couldn't these 3 numeric fields be an `union`, with `foreachName == NULL` as the discriminant? Not sure how much impact it would have, though.

Reply:
No; if it's a foreach context, it needs all three values, and if it's not, it still needs `foreachName == NULL`. There's no value that all contexts *except* foreach have to unify with the foreach-only value and step.

Reply:
(`FOREACH` *does* use `nbReptIters` to determine how many times it repeats; it calculates it at the initialization in `fstk_RunForeach`. That's why redefining the variable within the foreach body doesn't affect the behavior. It would be a different but also consistent model to store the current value, max, and step, instead of storing the current value, step, and count; but then `yywrap` would handle `FOREACH` differently.)